### PR TITLE
Dockerfile and minor updates

### PR DESCRIPTION
**Pull request by mgp25**
This PR will add a Dockerfile and minor updates.

```
docker run -it --rm -v $(pwd):/data depix -p images/testimages/testimage3_pixels.png -s images/searchimages/debruinseq_notepad_Windows10_closeAndSpaced.png -o /data/output.png

INFO:root:Loading pixelated image from images/testimages/testimage3_pixels.png
INFO:root:Loading search image from images/searchimages/debruinseq_notepad_Windows10_closeAndSpaced.png
INFO:root:Finding color rectangles from pixelated space
INFO:root:Found 116 same color rectangles
INFO:root:86 rectangles left after moot filter
INFO:root:Found 1 different rectangle sizes
INFO:root:Finding matches in search image
INFO:root:Removing blocks with no matches
INFO:root:Splitting single matches and multiple matches
INFO:root:[10 straight matches | 76 multiple matches]
INFO:root:Trying geometrical matches on single-match squares
INFO:root:[15 straight matches | 71 multiple matches]
INFO:root:Trying another pass on geometrical matches
INFO:root:[17 straight matches | 69 multiple matches]
INFO:root:Writing single match results to output
INFO:root:Writing average results for multiple matches to output
INFO:root:Saving output image to: /data/output.png
```

### CODE REVIEW

**Review by Saibamen — CHANGES_REQUESTED** [review]
Missing `.dockerignore`

Please use https://github.com/beurtschipper/Depix/blob/main/.gitignore file list

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM python:3
```
Comment (Saibamen) [inline Dockerfile:1]: Please use `python:3.9-slim`:

```suggestion
FROM python:3.9-slim
```

https://hub.docker.com/layers/python/library/python/3.9-slim/images/sha256-5d300921213569f955c3954787298127d241c115f31fc92e87305c3e7a71c7ef?context=explore

`python:3.9-slim` compressed size is 40.78 MB
`python:3` size is 325.36 MB

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+WORKDIR /current
+
+COPY . .
+
+RUN python -m pip install -r requirements.txt
+
+ENTRYPOINT ["python", "depix.py"]
```
Comment (Saibamen) [inline Dockerfile:9]: `CMD` will be better?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM python:3
+
+WORKDIR /current
+
+COPY . .
+
+RUN python -m pip install -r requirements.txt
```
Comment (Saibamen) [inline Dockerfile:7]: Consider usage of `--upgrade pip`

File: README.md [hunk 4]
```diff
@@ -26,3 +26,13 @@ The algorithm uses the fact that the linear box filter processes every block sep
 For most pixelized images Depix manages to find single-match results. It assumes these are correct. The matches of surrounding multi-match blocks are then compared to be geometrically at the same distance as in the pixelized image. Matches are also treated as correct. This process is repeated a couple of times.
 
 After correct blocks have no more geometrical matches, it will output all correct blocks directly. For multi-match blocks, it outputs the average of all matches.
+
+## Dockerfile
```
Comment (Saibamen) [inline README.md:30]:
```suggestion
## Docker
```

**Review by Saibamen — CHANGES_REQUESTED** [review]

**Comment by beurtschipper** [comment]
Thanks! I've created a separate docker branch because I'm not a big fan of docker, but I like the addition. I'll also add the changes to the argparser to the main branch.

**Comment by mgp25** [comment]
Thanks! 😄

**Comment by Saibamen** [comment]
@beurtschipper: Docker is must have in 2020...

Your project is open source. Just look at pull requests - 2 of them was about adding Docker support.

IMO better Docker support is in https://github.com/beurtschipper/Depix/pull/18 and https://github.com/beurtschipper/Depix/pull/18 should be merged into master branch

**Comment by toydotgame** [comment]
> @beurtschipper: Docker is must have in 2020...
> 
> Your project is open source. Don't force other developers to install everything in their local machines and mix their environments.
> Just look at pull requests - 2 of them was about adding Docker support.
> 
> IMO better Docker support is in #18 and #18 should be merged into master branch

Docker is such a bloat to Depix!
Running
```sh
python -m pip install -r requirements.txt`
```
...is _so much better_ than the 8 scripts, 10 additional files, and 12+ headaches from Docker.
Depix uses _**1**_ dependency; and it sounds like one that most people would already have!

**Pull request merged**